pkg/sql/conn_executor_prepare.go

```diff
@@ -344,14 +344,28 @@ func (ex *connExecutor) populatePrepared(
 func (ex *connExecutor) execBind(
 	ctx context.Context, bindCmd BindStmt,
 ) (fsm.Event, fsm.EventPayload) {
-	retErr := func(err error) (fsm.Event, fsm.EventPayload) {
+	retErr1 := func(err error) (fsm.Event, fsm.EventPayload) {
+		if bindCmd.PreparedStatementName != "" {
+			err = errors.Wrapf(err, "bind prepared statement %q failed", bindCmd.PreparedStatementName)
```

Comment:
nit: pls consider `errors.WithDetail` instead of copying the full SQL syntax as a prefix.

Reply:
👍